lib/chef/knife/vsphere_vm_network_add.rb

```diff
@@ -50,19 +60,65 @@ def run
     else fail
     end
 
-    vm.ReconfigVM_Task(
-      spec: {
-        deviceChange: [{
-          operation: :add,
-          fileOperation: nil,
-          device: RbVmomi::VIM::VirtualVmxnet3(
-            key: -1,
-            deviceInfo: { summary: summary, label: '' },
-            backing: backing,
-            addressType: 'generated'
-          )
-        }]
-      }
-    ).wait_for_completion
+    adapter_type = ''
+    if config[:adapter_type] == 'vmxnet3'
+      puts 'Adapter type: vmxnet3'
+      adapter_type = 'vmxnet3'
+    end
+    if config[:adapter_type] == 'e1000'
+      adapter_type = 'e1000'
+      puts 'Adapter type: e1000'
+    end
+    if config[:adapter_type].nil?
+      puts 'Adapter type: default=vmxnet3'
+      adapter_type = 'vmxnet3'
+    end
+
+    if config[:mac_address].nil?
```

Comment:
With a `default` in the option you could use `get_config(:mac_address)` right in the spec and then something like

```
GENERATED_MAC = '' # At the top
addressType: get_config(:mac_address) == GENERATED_MAC ? 'generated' : 'manual'
```

in the spec, too.